airflow-core/src/airflow/api_fastapi/core_api/routes/public/dag_sources.py

```diff
@@ -69,11 +71,14 @@ def get_dag_source(
             status.HTTP_404_NOT_FOUND,
             detail=f"Code not found. dag_id='{dag_id}' version_number='{version_number}'",
         )
+    file_location = dag_version.dag_code.fileloc
+    file_name = Path(file_location).name if file_location else None
```

Comment:
That should be in the response serializer. (Use aliases et computed fields to achieve the same)